packages/devtools_app/lib/src/standalone_ui/ide_shared/property_editor/property_editor_controller.dart

```diff
@@ -0,0 +1,47 @@
+// Copyright 2024 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:devtools_app_shared/utils.dart';
+import 'package:flutter/foundation.dart';
+
+import '../../../service/editor/api_classes.dart';
+import '../../../service/editor/editor_client.dart';
+
+class PropertyEditorController extends DisposableController
+    with AutoDisposeControllerMixin {
+  PropertyEditorController(this.editorClient) {
+    _init();
+  }
+
+  final EditorClient editorClient;
+
+  TextDocument? _currentDocument;
+  CursorPosition? _currentCursorPosition;
+
+  ValueListenable<List<EditableArgument>> get editableArgs => _editableArgs;
+  final _editableArgs = ValueNotifier<List<EditableArgument>>([]);
+
+  void _init() {
+    autoDisposeStreamSubscription(
+      editorClient.activeLocationChangedStream.listen((event) async {
+        final textDocument = event.textDocument;
+        final cursorPosition = event.selections.first.active;
+        // Don't do anything if the event corresponds to the current position.
+        if (textDocument == _currentDocument &&
+            cursorPosition == _currentCursorPosition) {
+          return;
+        }
+        _currentDocument = textDocument;
+        _currentCursorPosition = cursorPosition;
+        // Get the editable arguments for the current position.
+        final result = await editorClient.getEditableArguments(
```

Comment:
Is there anything to ensure this service was registered before it's called? Currently the method is prefixed with `experimental/` but that will change at some point in future. Since DevTools ships in the SDK, probably it'll be hard to get these out of sync, but it could certainly happen if running either the analysis server or DevTools from source.

There is an `initialized` event sent over DTD's `Lsp` stream to indicate when all LSP services have been registered, so if the registration has not happened before then, it isn't available.